app/src/main/java/com/danlls/daniel/todule_android/activities/TimeReminderFragment.java

```diff
@@ -148,21 +148,26 @@ public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
     public boolean onOptionsItemSelected(MenuItem item) {
         switch(item.getItemId()) {
             case R.id.action_save:
-                long timingInMillis = getOnlyHoursAndMins(Integer.parseInt(timeViewDays.getText().toString()),2)
-                        + getOnlyHoursAndMins(myCalendar.get(Calendar.HOUR_OF_DAY),1)
-                        + getOnlyHoursAndMins(myCalendar.get(Calendar.MINUTE),3);
-                MainActivity.timing = timingInMillis;
-                MainActivity.Tdays = Integer.parseInt(timeViewDays.getText().toString());
-                MainActivity.Thours = myCalendar.get(Calendar.HOUR_OF_DAY);
-                MainActivity.Tmins = myCalendar.get(Calendar.MINUTE);
-                SharedPreferences.Editor editor = sf.edit();
-                editor.putLong("timing", timingInMillis);
-                editor.putInt("hours", myCalendar.get(Calendar.HOUR_OF_DAY));
-                editor.putInt("minutes", myCalendar.get(Calendar.MINUTE));
-                editor.putInt("days", Integer.parseInt(timeViewDays.getText().toString()));
-                editor.commit();
-                ShowTos(timeViewDays.getText().toString()+" Days and "+timeViewHours.getText().toString());
-                return true;
+                if (!timeViewDays.getText().toString().equals("") && timeViewDays.getText().toString().matches("[0-9]+")) {
+                    long timingInMillis = getOnlyHoursAndMins(Integer.parseInt(timeViewDays.getText().toString()), 2)
+                            + getOnlyHoursAndMins(myCalendar.get(Calendar.HOUR_OF_DAY), 1)
+                            + getOnlyHoursAndMins(myCalendar.get(Calendar.MINUTE), 3);
+                    MainActivity.timing = timingInMillis;
+                    MainActivity.Tdays = Integer.parseInt(timeViewDays.getText().toString());
+                    MainActivity.Thours = myCalendar.get(Calendar.HOUR_OF_DAY);
+                    MainActivity.Tmins = myCalendar.get(Calendar.MINUTE);
+                    SharedPreferences.Editor editor = sf.edit();
+                    editor.putLong("timing", timingInMillis);
+                    editor.putInt("hours", myCalendar.get(Calendar.HOUR_OF_DAY));
+                    editor.putInt("minutes", myCalendar.get(Calendar.MINUTE));
+                    editor.putInt("days", Integer.parseInt(timeViewDays.getText().toString()));
+                    editor.apply();
+                    ShowTos("Reminder updated "+timeViewDays.getText().toString() + " Days and "
+                            + myCalendar.get(Calendar.HOUR_OF_DAY)+":"+myCalendar.get(Calendar.MINUTE));
```

Comment:
Since you're using days in the toast message, might as well use hours and minutes too instead of separating them using ":"